ajax.php

```diff
@@ -16,6 +16,40 @@
 //   You should have received a copy of the GNU Affero General Public License
 //   along with iTop. If not, see <http://www.gnu.org/licenses/>
 
+function MakeAttachmentLabel($oOrmDoc, $iObjId)
+{
+	$oAttachment = MetaModel::GetObject('Attachment', $iObjId);
+	$sTimeStampUpload = $oAttachment->Get('creation_date');
+	$sFileFormattedSize = $oOrmDoc->GetFormattedSize();
+	$sNameUploader = utils::HtmlEntities($oAttachment->Get('contact_id_friendlyname'));
+	$sAttachmentLabel = $oOrmDoc->GetFileName();
+	$sAttachmentLabel = utils::HtmlEntities($sAttachmentLabel);
+
+	$bHasUploaderName = $sNameUploader != '';
+	$bHasUploadTimestamp = $sTimeStampUpload != '';
+	if ($bHasUploaderName && $bHasUploadTimestamp)
+	{
+		$sDictEntryCode = 'UI-emry-attachment-label-case-A';
```

Comment:
```suggestion
		$sDictEntryCode = 'UI-emry-attachment-label-with-uploadername-and-timestamp';
```

Reply:
If you would name it `-with-uploadername-with-timestamp`, it could even be programatically easier to select the correct dict. (1 if statement less)

Reply:
But it would feel weird don't you think? Not sure it's worth it only remove 1 "if" statement.

Reply:
This looks much more clean, isn't it?

```php
	$sDictEntryCode = 'UI-emry-attachment-label';
	if ($bHasUploaderName) $sDictEntryCode .= '-with-uploadername';
	if ($bHasUploadTimestamp) $sDictEntryCode .= '-with-timestamp';
```

Reply:
Why not, but no one-line-if-condition (see conventions) :)

Reply:
Hmm, but this is allowed in the conventions:

```php
	$sDictEntryCode = 'UI-emry-attachment-label';
	sDictEntryCode .= ($bHasUploaderName) ? '-with-uploadername' : '';
	sDictEntryCode .= ($bHasUploadTimestamp) ? '-with-timestamp' : '';
```

Reply:
Yes.